roachtest: quit-all-nodes failed

[(roachtest).quit-all-nodes failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1860270&tab=buildLog) on [release-20.1@f5d82d1cdb75cfc3af9e35446d42c122050d9c9f](https://github.com/cockroachdb/cockroach/commits/f5d82d1cdb75cfc3af9e35446d42c122050d9c9f):

```
The test failed on branch=release-20.1, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20200408-1860270/quit-all-nodes/run_1
	quit.go:405,quit.go:434,quit.go:145,context.go:135,quit.go:144,quit.go:434,test_runner.go:753: error with attached stack trace:
		    main.execCmdWithBuffer
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:426
		    main.(*cluster).RunWithBuffer
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2041
		    main.runQuit
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/quit.go:402
		    main.registerQuitAllNodes.func1.1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/quit.go:434
		    main.(*quitTest).runWithTimeout.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/quit.go:145
		    github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:135
		    main.(*quitTest).runWithTimeout
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/quit.go:144
		    main.registerQuitAllNodes.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/quit.go:434
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:753
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - error with embedded safe details: %s
		    -- arg 1: <string>
		  - /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1860270-1586330972-18-n5cpu4:5 -- ./cockroach quit --insecure --logtostderr=INFO --port={pgport:5} --drain-wait=1h:
		  - exit status 30

```

<details><summary>More</summary><p>


Artifacts: [/quit-all-nodes](https://teamcity.cockroachdb.com/viewLog.html?buildId=1860270&tab=artifacts#/quit-all-nodes)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Aquit-all-nodes.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).quit-all-nodes failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1862701&tab=buildLog) on [release-20.1@9d520536d4273264c58494222401b682d88d7edd](https://github.com/cockroachdb/cockroach/commits/9d520536d4273264c58494222401b682d88d7edd):

```
The test failed on branch=release-20.1, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20200409-1862701/quit-all-nodes/run_1
	quit.go:405,quit.go:434,quit.go:145,context.go:135,quit.go:144,quit.go:434,test_runner.go:753: error with attached stack trace:
		    main.execCmdWithBuffer
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:426
		    main.(*cluster).RunWithBuffer
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2041
		    main.runQuit
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/quit.go:402
		    main.registerQuitAllNodes.func1.1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/quit.go:434
		    main.(*quitTest).runWithTimeout.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/quit.go:145
		    github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:135
		    main.(*quitTest).runWithTimeout
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/quit.go:144
		    main.registerQuitAllNodes.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/quit.go:434
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:753
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - error with embedded safe details: %s
		    -- arg 1: <string>
		  - /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1862701-1586418317-17-n5cpu4:5 -- ./cockroach quit --insecure --logtostderr=INFO --port={pgport:5} --drain-wait=1h:
		  - exit status 30

```

<details><summary>More</summary><p>


Artifacts: [/quit-all-nodes](https://teamcity.cockroachdb.com/viewLog.html?buildId=1862701&tab=artifacts#/quit-all-nodes)
Related:
- #47183 roachtest: quit-all-nodes failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-master](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-master) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Aquit-all-nodes.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/cmd/roachtest/quit.go
+++ b/pkg/cmd/roachtest/quit.go
@@ -443,6 +443,11 @@ func registerQuitAllNodes(r *testRegistry) {
 			// also expect the node to exit successfully.
 			q.runWithTimeout(ctx, func(ctx context.Context) { expectHardShutdown(ctx, q.t, runQuit(ctx, q.t, q.c, 2, "--drain-wait=4s")) })
 			q.runWithTimeout(ctx, func(ctx context.Context) { expectHardShutdown(ctx, q.t, runQuit(ctx, q.t, q.c, 1, "--drain-wait=4s")) })
+
+			// At the end, restart all nodes. We do this to check that
+			// the cluster can indeed restart, and also to please
+			// the dead node detection check at the end of each test.
+			q.c.Start(ctx, q.t, q.args)
 		},
 	})
 }

